data/src/main/java/org/apache/iceberg/data/TableMigrationUtil.java

```diff
@@ -140,28 +141,36 @@ public static List<DataFile> listPartition(
             index -> {
               Metrics metrics = getAvroMetrics(fileStatus.get(index).getPath(), conf);
               datafiles[index] =
-                  buildDataFile(fileStatus.get(index), partitionValues, spec, metrics, "avro");
+                  metrics.recordCount() > 0
+                      ? buildDataFile(fileStatus.get(index), partitionValues, spec, metrics, "avro")
```

Comment:
Instead of adding nulls, how about we filter out the files in `filesStatus` that are empty?

Reply:
It seems to me there is no such good way. The number of records is not stored within `filesStatus`, it is stored in `metrics` which is obtained via a I/O call by each parallelized task executors. We don't know which files are really empty before entering the task execution.

Reply:
Is `filestatus.getLen()` not helpful?

Reply:
it is not helpful, because we should not assume that empty record file in different formats will have the static`filestatus.getLen()` value.